WickedEngine/wiRenderer.cpp

```diff
@@ -2385,6 +2385,36 @@ ForwardEntityMaskCB ForwardEntityCullingCPU(const Visibility& vis, const AABB& b
 	return cb;
 }
 
+void Workaround(const int bug , CommandList cmd)
+{
+	if (device->GetBackend() == BACKEND_DX12)
+	{
+		if (bug == 1)
+		{
+			//PE: Strange DX12 bug, we must change the pso/pipeline state, just one time.
+			//PE: After this there will be no "black dots" or culling/depth errors.
+			//PE: This bug only happen on some nvidia cards ?
+			//PE: https://github.com/turanszkij/WickedEngine/issues/450#issuecomment-1143647323
+
+			//PE: We MUST use RENDERPASS_VOXELIZE (DSSTYPE_DEPTHDISABLED) or it will not work ?
+			const PipelineState* pso = &PSO_object[0][RENDERPASS_VOXELIZE][BLENDMODE_OPAQUE][0][0][0];
+			
+			static RenderPass renderpass_clear;
+			if (!renderpass_clear.IsValid())
+			{
+				RenderPassDesc renderpassdesc;
+				renderpassdesc.flags = RenderPassDesc::Flags::EMPTY;
+				device->CreateRenderPass(&renderpassdesc, &renderpass_clear);
+			}
+			device->RenderPassBegin(&renderpass_clear, cmd);
+			device->BindPipelineState(pso, cmd);
+			device->DrawIndexedInstanced(0, 0, 0, 0, 0, cmd); //PE: Just need predraw(cmd);
```

Comment:
Do you know which part of the predraw() is needed? Does it need the PSO, or maybe just a root signature or root parameter?

Reply:
Yes the pso is needed and it must be RENDERPASS_VOXELIZE , i tried a lot others they dont fix it, perhaps its the DSSTYPE_DEPTHDISABLED that do it ? but i tried other debug PSO with DSSTYPE_DEPTHDISABLED but they did not fix it, so it must be a combination in RENDERPASS_VOXELIZE ?. Anyway its needed.